src/python/pants/backend/go/target_type_rules.py

```diff
@@ -46,6 +60,36 @@ async def inject_go_package_dependencies(
         return InjectedDependencies()
 
 
+# TODO: Figure out how to merge (or not) this with ResolvedImportPaths as a base class.
+@dataclass(frozen=True)
+class GoImportPathToPackageMapping:
+    # Maps import paths to the address of go_package or (more likely) _go_ext_mod_package targets.
+    mapping: FrozenDict[str, tuple[Address, ...]]
+
+
+@rule
+async def analyze_import_path_to_package_mapping() -> GoImportPathToPackageMapping:
+    mapping: dict[str, list[Address]] = {}
```

Comment:
`defaultdict(list)` can make this much nicer. You say `mapping[import_path].append(tgt.address)` and it will create the key for you if it doesn't yet exist.

Reply:
Fixed.